parlai/agents/seq2seq/hred.py

```diff
@@ -0,0 +1,397 @@
+#!/usr/bin/env python3
+import torch
+import torch.nn as nn
+
+from .modules import (
+    AttentionLayer,
+    OutputLayer,
+    RNNEncoder,
+    Seq2seq,
+    _transpose_hidden_state,
+    opt_to_kwargs,
+)
+from .seq2seq import Seq2seqAgent
+
+
+class HredModel(Seq2seq):
```

Comment:
Why are we inheriting from seq2seq, when many of the options I think we don't support?

Reply:
Ah, I structured it like this because it originally contained most of this functionality--wasn't sure if it was going to be added back in in some form later--I will switch it to TorchGeneratorModel and update the PR. Thanks!